rero_ils/modules/imports/api.py

```diff
@@ -395,97 +413,90 @@ def search_records(self, what, relation, where='anywhere', max_results=0,
-                                .find('{*}numberOfRecords').text
-                            )
-
+                    self.cache.setex(
+                        cache_key,
+                        timedelta(minutes=self.cache_expire),
+                        value=pickle.dumps(cache_data)
+                    )
             self.results['hits']['total']['value'] = len(
                 self.results['hits']['hits'])
-            if self.results['hits']['total']['value'] == 0:
-                self.status_code = 404
-                self.results['errors'] = {
-                    'code': self.status_code,
-                    'title': f'{self.name}: Not found: '
-                             f'{where} {relation} {what}'
-                    }
-            else:
-                cache_data = {
-                    'hits': self.results['hits'],
-                    'data': self.data
-                }
-                self.cache.setex(
-                    cache_key,
-                    timedelta(minutes=self.cache_expire),
-                    value=pickle.dumps(cache_data)
-                )
-                self.create_aggregations(self.results)
-                self.status_code = 200
-        except Timeout as error:
-            current_app.logger.warning(f'{self.name}: {error}')
-            # abort(503, description='Timeout')
+            self.create_aggregations(self.results)
+        except requests.exceptions.ConnectionError as error:
+            self.status_code = 433
+            self.status_msg = str(error)
+        except requests.exceptions.HTTPError as error:
+            current_app.logger.error(f'{type(error)} {error}')
+            self.status_code = error.response.status_code
+            self.status_msg = str(error)
+            import traceback
+            current_app.logger.error(f'HTTPError: {traceback.format_exc()}')
         except Exception as error:
+            self.status_code = 500
+            self.status_msg = str(error)
+            import traceback
+            current_app.logger.error(f'Exception: {traceback.format_exc()}')
+        if self.status_code > 400:
+            # TODO: enable error logging only for 500
+            # if self.status_code == 500:
             current_app.logger.error(
-                f'{type(error).__name__} {self.name}: {error}')
-            abort(500, description='Error')
+                f'Import: {self.name} '
+                f'code: {self.status_code} '
+                f'error: {self.status_msg} '
+                f'url: {url_api}'
+            )
+            self.results['errors'] = {
+                'code': self.status_code,
+                'message': self.status_msg,
+                'url': url_api
+            }
         return self.results, self.status_code
 
 
 class BnfImport(Import):
     """Import class for BNF."""
 
     name = 'BNF'
-    url = 'http://catalogue.bnf.fr'
+    url = 'https://catalogue.bnf.fr'
```

Comment:
Is this should be moved in the configuration one day?

Reply:
Maybe, but this is already an class for BNF import. normally this URL should not change.